roles/system/templates/services/system-sphinx.yaml

```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ sphinx.env.thinking.service | default(sphinx_env_thinking_service) }}
```

Comment:
I don't understand why the name of the k8s `Service` name and even the port are not harcoded, while:
- I have seen on some deployments, the value of the env var is harcoded to `system-sphinx`, while on other places is taken from CR  env var (with a default value `system-sphinx`)
- The same for the port, on some places like container port or readiness probe is harcoded to 9306, and on other places is used the env var
- In addition, host/port CR env vars are used to create a configmap, and then this configmap is mounted to extract env vars on some deployments
- I don't think we gain something by making host, port, bind-addres, path, pid-file... customizable via CR (other deployment has harcoded mountpaths for example). From my point of view only **full and delta reindex intervals** are things we may require to customize, with that change sphinx config would be much more simple

What do you think?

Reply:
There are 2 ports, the service port where the service is exposed and the sphinx actual port. The one that is extracted as an env var is the first one, used by the system app and sidekiq deployments.

As we discussed over slack, there have been some changes since the review that tackle some of the comments. Instead of using the service var, now defaults to the hardcode "system-sphinx" and the `service` attribute has been changed to `endpoint` (and it's optional). Before removing the parametrisation already available in the current version, I will try to speak with @3scale/system how those env vars are used and decide the best (and cleanest) approach.

Any how, the `sphinx` rework may be something to work in another PR.

Sounds good?